zkvm/docs/zkvm-spec.md

```diff
@@ -371,6 +374,42 @@ Time bounds are available in the transaction as [expressions](#expression-type)
 [`mintime`](#mintime) and [`maxtime`](#maxtime).
 
 
+### Contract ID
+
+_Contract ID_ is a 32-byte unique identifier of the [contract](#contract-type) and a commitment to its contents (see [Output](#output-structure)):
+
+```
+T = Transcript("ZkVM.contractid")
+T.commit("contract", output_structure)
+id = T.challenge_bytes("id")
+```
+
+Contract ID makes signatures safe against reused [predicate](#predicate) keys:
```

Comment:
What does the phrase "predicate key" mean here?

Reply:
Changed to 
```
Contract ID makes signatures safe against reused [verification keys](#verification-key) in [predicates](#predicate):
```